tests/moe_ep/test_moe_ep_mxfp8_cutedsl_mega_multirank.py

```
@@ -555,6 +564,219 @@ def test_moe_ep_mxfp8_cutedsl_mega_layer_large_tokens_matches_reference():
+
+        assert torch.isfinite(y_kernel).all()
+        yk = y_kernel.to(torch.float32)
+        rel_l2 = (yk - y_ref).norm() / y_ref.norm().clamp_min(1e-6)
+        print(
+            f"[mxfp8 multirank oracle rank {rank} ikr={in_kernel_fc2_reduce}] "
+            f"rel_l2={rel_l2.item():.4g} "
+            f"max|d|={(yk - y_ref).abs().max().item():.4g} "
+            f"amax(ref)={y_ref.abs().max().item():.4g}"
+        )
+        # Single-GPU oracle tolerances: random unscaled bf16 weights yield
+        # |y|~1e2–1e3; kernel vs torch ref can differ by ~1 bf16 ULP (|d|≈8)
```

Ruff (0.16.0)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)